sale_stock_refund_picking/models/sale_order.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.multi
+    def _get_delivered_qty(self):
+        """Deduct moves marked to refund."""
+        qty = super(SaleOrderLine, self)._get_delivered_qty()
+        for move in self.procurement_ids.mapped('move_ids').filtered(
+                lambda r: (r.state == 'done' and
```

Comment:
Uhm, curious way to get refunds... This doesn't serve for purchases as returned doesn't preserve procurement group I think.